subprojects/dependency-management/src/main/java/org/gradle/api/internal/artifacts/configurations/DefaultConfiguration.java

```diff
@@ -1597,7 +1627,7 @@ public List<String> getResolutionAlternatives() {
     }
 
     @Override
-    public boolean isFullyDeprecated() {
+    public boolean isFullyDeprecated() { // TODO: Should this include canBeDeclared?
```

Comment:
Looks like this method is being used in three places. 

The first is `DefaultArtifactHandler`. In reality, I believe that method should be checking `configuration.getDeclarationAlternatives() != null`, similar to how `DefaultDependencySet` and `DefaultDependencyConstraintSet` are doing that already. 

The second is `AbstractDependencyReportTask`. This task displays a list of dependencies declared on a given configuration. So, in reality, we want this to check `configuration.canBeDeclared() == true`. 

The last usage is `JsonProjectDependencyRenderer`, which has a similar use as above, except it prints to JSON instead of the console. Again, I think we should instead check `configuration.canBeDeclared() == true` since this code is looking for any configuration with dependencies. 

In general, I think this method should be removed entirely.